provider/oci/environ.go

```diff
@@ -393,10 +406,16 @@ func (e *Environ) DestroyController(ctx envcontext.ProviderCallContext, controll
 
 	err = e.StopInstances(ctx, ids...)
 	if err != nil {
+		providerCommon.HandleCredentialError(err, ctx)
 		return err
 	}
 	logger.Debugf("Cleaning up network resources")
-	return e.cleanupNetworksAndSubnets(controllerUUID, "")
+	err = e.cleanupNetworksAndSubnets(controllerUUID, "")
+	if err != nil {
```

Comment:
No need for nil check, handle that in the helper func if necessary

return providerCommon.HandleCredentialError(err, ctx)